admin-auth/src/main/kotlin/com/epam/drill/admin/auth/config/ApiKeyAuthenticationProvider.kt

```diff
@@ -86,7 +97,9 @@ fun Authentication.Configuration.apiKey(
 
     provider.pipeline.intercept(AuthenticationPipeline.RequestAuthentication) { context ->
         val apiKey = context.call.request.header(provider.headerName)
-        val principal = apiKey?.let { provider.authenticationFunction(context.call, it) }
+
+        val principal = runCatching { provider.cache.getIfPresent(apiKey) }.getOrNull()
```

Comment:
- `runCatching`: It is better to avoid catching situations that can be avoided by a simple check. TBD
- Need to think about how to disable the cache

Reply:
refactored